apps/condo/domains/meter/schema/RegisterMetersReadingsService.js

```diff
@@ -415,20 +459,18 @@ const RegisterMetersReadingsService = new GQLCustomSchema('RegisterMetersReading
                                 isAutomatic: get(reading, ['meterMeta', 'isAutomatic']),
                             })
                             meterId = createdMeter.id
+                            meters.push(transformToPlainObject(createdMeter))
                         }
                     } catch (e) {
                         resultRows.push(e)
                         continue
                     }
 
                     try {
-                        const duplicates = await MeterReading.getAll(context, {
-                            meter: { id: meterId },
-                            date: toISO(reading.date),
-                            ...values,
-                        })
+                        const key = `${meterId}-${dateISO}`
+                        const duplicateReading = meterReadingByDate[key]
 
-                        if (duplicates.length === 0) {
+                        if (!duplicateReading) {
```

Comment:
No, because if there is no duplicate, reading will be created and set in `meterReadingByDate`. And for create reading, we need a existed meter, which created above